h/views/account_signup.py

```diff
@@ -343,12 +349,47 @@ def encode_idinfo_token(
     provider_unique_id: str,
     issuer: JWTIssuer,
     audience: JWTAudience,
+    next_url: str,
 ):
     return {
         "idinfo": jwt_service.encode_symmetric(
-            IDInfo(provider_unique_id),
+            IDInfo(provider_unique_id, next_url=next_url),
             expires_in=timedelta(hours=1),
             issuer=issuer,
             audience=audience,
         ),
     }
+
+
+def redirect_url(request, user, url):
```

Comment:
Should probably add a docstring to this. We don't want to redirect to just any old `next` URL that we find in a query param (or in a `state` or `idinfo` token). That could be a vector for attacks. We'll only redirect to the `/oauth/authorize` URL. Anything else is just ignored and we redirect to the user profile page instead.

Reply:
Done